dataset/histogram.cpp

```diff
@@ -64,84 +77,95 @@ static constexpr auto make_histogram =
           }
         }
       }
-    };
-
-static constexpr auto make_histogram_unit = [](const units::Unit &events_unit,
-                                               const units::Unit &weights_unit,
-                                               const units::Unit &edge_unit) {
-  if (events_unit != edge_unit)
-    throw except::UnitError("Bin edges must have same unit as the events "
-                            "input coordinate.");
-  if (weights_unit != units::counts && weights_unit != units::dimensionless)
-    throw except::UnitError("Weights of event data must be "
-                            "`units::counts` or `units::dimensionless`.");
-  return weights_unit;
-};
+    },
+    [](const units::Unit &events_unit, const units::Unit &weights_unit,
+       const units::Unit &edge_unit) {
+      if (events_unit != edge_unit)
+        throw except::UnitError("Bin edges must have same unit as the events "
```

Comment:
I don't think these unit errors are tested for

Reply:
Indeed. The plan is to extract the lambda from the source so it can be tested stand-alone, just like we did for the operations for variables in `core::element`.